paseos/tests/custom_property_test.py

```diff
@@ -0,0 +1,47 @@
+"""This test checks whether power charging is performed correctly"""
+
+from test_utils import get_default_instance
+
+import paseos
+from paseos import ActorBuilder, SpacecraftActor
+
+import pykep as pk
+import numpy as np
+
+
+def test_custom_power_consumption_property():
+    """Checks whether we can create a custom property to track power consumption of an actor"""
+
+    # Initialize the actor
+    sim, sat1, earth = get_default_instance()
+    sat1 = ActorBuilder.get_actor_scaffold("sat1", SpacecraftActor, pk.epoch(0))
+    ActorBuilder.set_orbit(sat1, [10000000, 0, 0], [0, 8000.0, 0], pk.epoch(0), earth)
+    ActorBuilder.set_power_devices(sat1, 50000, 1000000, 1, paseos.PowerDeviceType.SolarPanel)
+
+    # Add a custom property which tracks the total power consumption
+    prop_name = "total_power_consumption"
+
+    # Define the update function
+    def prop_update_fn(actor, dt, power_consumption):
```

Comment:
How do you specify "power_consumption"? This is not clear to me.

Reply:
The same way as always. Either you run an activity with defined power consumption 

```py
# Register an activity that emulate event detection
sim.register_activity(
    "activity_A",
    activity_function=activity_function_A,
    power_consumption_in_watt=10
)

#Run the activity
sim.perform_activity("activity_A")
```

or you call 

```py
sim.advance_time(3600, power_consumption=10)
```

Nothing  changed. :)